compiler/rustc_middle/src/mir/mod.rs

```diff
@@ -1705,6 +1710,12 @@ impl<'tcx> PlaceRef<'tcx> {
         }
     }
 
+    #[inline]
+    pub fn iter_subplaces(self) -> impl Iterator<Item = PlaceRef<'tcx>> {
```

Comment:
Please add a doc comment, in particular comparing with `iter_projections`. Crucially this iterator here is not empty on a place without projections!